lib/config_skeleton.rb

```diff
@@ -188,6 +188,19 @@ def initialize(env)
     end
 
     initialize_config_skeleton_metrics
+    @trigger_regen_r, @trigger_regen_w = IO.pipe
+  end
+
+  # Expose the write pipe which can be written to to trigger a config
+  # regeneration with a forced reload; a similar mechanism is used for
+  # shutdown but in that case writes are managed internally.
+  #
+  # Usage: config.reload_trigger.write(".") . It does not matter what
+  # is written, we are only detecting that something was written.
+  #
+  # @return [IO]
+  def regen_trigger
+    @trigger_regen_w
```

Comment:
I think the API should be simpler: 

```
def notify_config_change
   @trigger_regen_w << "x"
end
```

Reply:
@SamSaffron I can't do this because I need to be able to pass some sort of object all the way down to the observer so it can signal that it needs the config regen. I agree that it is not nice to have to know to `.write(".")`

Reply:
Oh then wrap it with an object ... 

```
class ConfigChangeNotifier
   def notify_change
       ...
   end
end
```

Simpler to reason about how to do the notifications and the pipe is abstracted away.

Reply:
you could also pass `self` around?

Reply:
The other thing I could think of is making a small class and passing the instance variable into that, which has a single method of `regen!`, and passing an instance of that down to the observer.

Reply:
This is so weird it’s like my comment was posted at the same time as yours...will use a little object like this, passing self seems excessive because the config class is huge